TranslatorReasonerAPI.yaml

```diff
@@ -1008,6 +1043,51 @@ components:
       required:
         - subject
         - object
+    QPath:
+      additionalProperties: true
+      description: A path in the QueryGraph used for pathfinder queries only, flowing
+        from subject to intermediate nodes, if any, to object.
+      properties:
+        subject:
+          description: Corresponds to the map key identifier of the subject concept
+            node for the start of the queried path.
+          example: n0
+          type: string
+        object:
+          description: Corresponds to the map key identifier of the object concept
+            node for the end of the queried path.
+          example: n1
+          type: string
+        intermediate_nodes:
+          default: []
+          description: An ordered list of intermediate nodes, if any, in the direction of
+            subject to object in the query path.
+          items:
+            type: string
+          type: array
+        predicates:
+          description: These should be Biolink Model predicates and are allowed to
+            be of type 'abstract' or 'mixin' (only in QGraphs!). Use of 'deprecated'
+            predicates should be avoided.
+          items:
+            $ref: '#/components/schemas/BiolinkPredicate'
+          minItems: 1
+          nullable: true
+          type: array
+        attribute_constraints:
+          default: []
+          description: A list of attribute contraints applied to a query edge. If
+            there are multiple items, they must all be true (equivalent to AND)
+          items:
+            $ref: '#/components/schemas/AttributeConstraint'
+          type: array
+      required:
+      - intermediate_nodes
```

Comment:
This might not be a required field. We only need intermediate nodes on constrained queries, so not all pathfinder queries require them.

Reply:
The default is [ ], an empty list. I idea is that there should not be two ways to say there are no intermediate nodes: null or [ ]. Rather there should be one way: [ ]. Thus we make it required and require an empty list to say there are none. This is a custom that we have been moving to in other parts of the TRAPI schema, although I don't feel strongly.

Reply:
The required empty lists to say "none" bloat the TRAPI response without adding any data. I think requiring properties when their value is "null" also bloats the TRAPI response. 

So I think this custom should be changed to having optional properties only - so tools don't need to provide the property if it's not required and they don't use it (aka remove from `required` section, don't use `nullable`).

Reply:
I tend to agree with @colleenXu ; It's a slight bit of extra consideration when parsing the TRAPI, but the bloat reduction is real. As long as everyone is playing on the same assumptions (which are given by descriptions in the TRAPI spec), I don't see a problem with it.

Reply:
I agree that this discussion of required empty lists should be moved to a diff ticket and shouldn't be a barrier to getting this PR done. 

(I do wonder if it can be added to 1.6, if we are all in agreement and it's an easy, backwards-compatible change to make...)